readthedocs/proxito/tests/responses/v1.json

```diff
@@ -155,7 +155,8 @@
       "inject_styles": true
     },
     "filetreediff": {
-      "enabled": false
+      "enabled": false,
+      "ignored_files": null
```

Comment:
A little surprised we're passing this to the frontend, instead of just filtering it in the API response?

Reply:
I decided to filter this out in the frontend to avoid executing a bunch of regex provided by the user in the server side. I think it's safer to do it in the client.

Reply:
Running random regex on the server is dangerous, that's why we use the regex lib for automation rules, which offers a timeout when evaluating the regex. So we are already running untrusted regex in the server with a timeout.

We could also use file patterns instead of regex as we do for search, which I think are good enough for this use case.

Filtering on the client is more secure, but can still result on random freezes if the user writes an inefficient regex, and we also need to note that the regex they input are js flavored. So I think filtering using file patterns should be better, maybe we can have an option in the future to return all files, even if ignored (show ignored (x), and have a live filtering instead of big list of files).

Reply:
I agree, I don't really understand why we need regex here? I think globbing should be plenty, and what we're already using for `search.ranking`: https://github.com/readthedocs/readthedocs.org/blob/5bce6ddab35f551e0adf9a864e9252ea4d8f70bc/readthedocs/projects/tasks/search.py#L68

Having regexs be slow on the client is still bad UX. We should avoid the problem completely, not just punt it to the client.

Reply:
Cool. I used regex because it was what I had at hand without doing too much research. I'm fine using using `fnmatch` as we are doing with search. In that case, I will perform the filtering of files in the server then, since we have the code already there for that.